pilot/cmd/pilot-agent/status/ready/probe.go

```diff
@@ -51,14 +52,14 @@ func (p *Probe) checkConfigStatus() error {
 		return err
 	}
 
-	CDSUpdated := s.CDSUpdatesSuccess > 0
-	LDSUpdated := s.LDSUpdatesSuccess > 0
-	if CDSUpdated && LDSUpdated {
-		p.receivedFirstUpdate = true
-		return nil
+	if s.CDSUpdatesSuccess == 0 || s.LDSUpdatesSuccess == 0 {
+		return fmt.Errorf("config not received from Pilot (is Pilot running?): %s", s.String())
 	}
-
-	return fmt.Errorf("config not received from Pilot (is Pilot running?): %s", s.String())
+	if p.NodeType == model.SidecarProxy && p.SDSEnabled && s.SDSUpdatesSuccess == 0 {
```

Comment:
won't gateway envoy also needs SDS updates in order to serve CDS configured with mTLS?

Reply:
gateway will send mTLS with default installation starting from 1.5. default is autoMTLS && permissive in the mesh.

Reply:
Correct. But if the user turns off mTLS, this stat check (without checking mTLS) will block the gateway pod from being brought up. So we can't really do anything for the gateway unless we also take the mTLS enablement factor.

Reply:
do i miss something, is it because some config difference ensuring sidecar always have a SDS update stats no matter what but not for gateway, and why? whether mTLS is enabled are the same to sidecar gateway.

Reply:
e.g. if you disable mTLs for entire mesh, and apply this PR for both gateway and sidecar, can sidecar envoy come up but not the gateway?

Reply:
> The 15006 listener is different. I found it's always retrieving secrets regardless of mTLS enablement.

Sorry I missed this. Got it. Thanks.

Could you leave a comment in the code to state why only for sidecar? Something like `For sidecar 15006 always requests secret and gateawy we can not ensures SDS requests will be sent from Envoy.` (feel free reword).